manifests/init.pp

```diff
@@ -58,6 +60,7 @@
   Variant[Enum[''], Stdlib::Absolutepath] $ssl_cert = $puppetserver_foreman::params::client_ssl_cert,
   Variant[Enum[''], Stdlib::Absolutepath] $ssl_key = $puppetserver_foreman::params::client_ssl_key,
   Boolean $use_client_tls_certs = true,
+  Boolean $watch_for_facts_and_publish_them = false,
```

Comment:
I don't know how many people will use foreman with PE. I assume the majority of FOSS users will configure foreman as ENC, the PE users will probably stick to the PE Console as ENC. So we could set `watch_for_facts_and_publish_them` to true on PE ans false on FOSS.

Reply:
At some point I thought hard about this bit never got to a satisfying answer. I know puppetlabs-pe_tools contains a derivative of this code (which I question is allowed by the license). I welcome thought on how it should work from someone who uses it in the real world

Reply:
It's not enabled by default on PE and disabled on open source.